upkie/cpp/actuation/BulletInterface.cpp

```diff
@@ -353,6 +361,41 @@ void BulletInterface::read_joint_sensors() {
     result.torque = sensor_state.m_jointMotorTorque;
   }
 }
+void BulletInterface::save_nominal_masses() {
+  const int nb_links = bullet_.getNumJoints(robot_);
+  b3DynamicsInfo info;
+  for (int link_id = 0; link_id < nb_links; ++link_id) {
+    bullet_.getDynamicsInfo(robot_, link_id, &info);
+    nominal_masses[link_id] = info.m_mass;
+    nominal_inertia[link_id][0] = info.m_localInertialDiagonal[0];
+    nominal_inertia[link_id][1] = info.m_localInertialDiagonal[1];
+    nominal_inertia[link_id][2] = info.m_localInertialDiagonal[2];
+  }
+}
+void BulletInterface::randomize_masses() {
+  for (const auto& link_id : nominal_masses) {
+    std::default_random_engine generator;
+    std::uniform_real_distribution<double> distribution(-inertia_randomization_,
+                                                        inertia_randomization_);
```

Comment:
Maybe move the generator and distribution out of the loop?

If I understand correctly we can allocate them once (outside the loop) then use them multiple times.